### ISSUE TITLE
geom_sina has problems with artificial data

### ISSUE THREAD

**Issue by idot** [thomasp85/ggforce#48]
I tried to test some things and geom_violin copes with my stupid test data, but geom_sina throws errors. Maybe thats indicative of some problems - I don't really know.

```R
df <- data.frame(g=c("A","B"),v=rep(c(rep(5,20),rep(10,20),rep(15,20)),2))
ggplot(df, aes(x=1,y=v)) + geom_violin()
ggplot(df, aes(x=1,y=v)) + geom_sina() //Error in f(...) : could not find function "has_groups"
ggplot(df, aes(x=g,y=v)) + geom_violin()
ggplot(df, aes(x=g,y=v)) + geom_sina() //Warning: Computation failed in `stat_sina()`: 'vec' must be sorted non-decreasingly and not contain NAs
```

**Comment by thomasp85** [comment]
The first one has been fixed in the devel version of ggforce - I'll look into the second...

**Comment by thomasp85** [comment]
@sidiropoulos do you have time to look into this as well?

**Comment by sidiropoulos** [comment]
@idot: Like @thomasp85 says, the first issue should be resolved if you install the devel version of geforce. 
I've reproduced the second error. A temporary fix would be to try
```
df <- data.frame(g = c("A", "B"), v = rep(c(rep(5, 20), rep(10, 20), rep(15, 20)), 2))  
ggplot(df, aes(x = 1, y = v + 0.0001)) + geom_sina()
```

You can add more (or less) zeros. The choice in the example above is arbitrary.
It's a break in the `findInterval()` function which I will fix and submit.

Thanks for reporting this!

**Comment by thomasp85** [comment]
Fixed in #52
